router/xgress_edge_transport/connection.go

```diff
@@ -143,7 +143,7 @@ func (c *edgeTransportXgressConn) Write(p []byte) (n int, err error) {
 		//if crypto enabled and we have not setup e2e
 		if c.cryptoCtx.rxKey != nil {
 			if len(p) != secretstream.StreamHeaderBytes {
-				return 0, fmt.Errorf("error establishing crypto: expected key length %d got %d", len(p), secretstream.StreamHeaderBytes)
+				return 0, fmt.Errorf("error establishing crypto: expected key length %d got %d", secretstream.StreamHeaderBytes, len(p))
```

Comment:
These values look swapped.

Reply:
No, it is correct as I have it.  We expect secretstream.StreamHeaderBytes (which is defined here: https://github.com/golang/crypto/blob/master/chacha20poly1305/chacha20poly1305.go#L28))

It was actually swapped before this.  We witnessed this mis-formatting issue in the field a while back, and this fix has been pending in my branch.